### ISSUE TITLE
Improve messaging for custom email setup

### ISSUE THREAD

**Issue by taipeicoder** [Automattic/wp-calypso#95931]
### Quick summary

The process of purchasing and setting up email is technical, the empty state doesn't do a great job at describing what users need to do. For instance, it says "upgrade to a plan now" but plans aren't really necessary to setting up email.

![Image](https://github.com/user-attachments/assets/5a277a80-fabb-404a-90ec-103dec15097a)

### Steps to reproduce

Head to /mail/:domain

### What you expected to happen

Clear guidance in the empty state.

### What actually happened

The empty state prioritizes plan upsell.

### Impact

One

### Available workarounds?

There is no user impact

**Comment by Robertght** [comment]
I was able to replicate this with a freshly new website, but, I believe this is a bug so I'm flagging it like so.

Setting this as a high priority.

I'm flagging this for @Automattic/yolo

**Comment by richtabor** [comment]
> Buttons: Get a domain / Use a domain I own

If a user has a plan, I think it'd be a good idea to help them set up their domain faster by providing a link to both. 

I created this issue to add a call to action for using an existing domain when users have plans: https://github.com/Automattic/wp-calypso/issues/98168

**Comment by StevenDufresne** [comment]
> If no plan and no domain:

I looked into this proposal which I think is good, but it doesn't appear like we have any flows in that order unless I'm mistaken. Based on what I see in the code, I'm not convinced it's worth the effort. The flow itself needs improvements that will have a bigger impact.

What I think we can do, is update the copy and move the user through the `domains/add` flow, which goes as follows:

1. Choose Domain
2. Add professional email
3. Checkout

Here's how it would look:

<img width="500" src="https://github.com/user-attachments/assets/607263a8-9668-4d4e-89d2-2b811442568f">
<img width="500" src="https://github.com/user-attachments/assets/196f3bf4-b260-4752-a3e3-a90c3332ef3b">
<img width="500" src="https://github.com/user-attachments/assets/12059c6f-6776-4bb7-b268-9ef215c9a240">
<img width="500" src="https://github.com/user-attachments/assets/5509a3f1-b850-4382-84e0-6e3af4586167">

We'd update the copy to something like:

#### If no plan and no domain:
Heading: `Add a professional custom email address`
Paragraph: `Choose a domain and set up a custom email address that truly represents you.`
Button: `Buy a domain and email`
<br>
<hr>
<br>

If we wanted to promote all plans, we could use the Just in time message flow at `/domains/add/{site_url}?domainAndPlanPackage=true` which has a slightly different UX but shows all the hosting plans instead of just email.

<img width="500" src="https://github.com/user-attachments/assets/149f1aee-91e9-4128-8591-f3448eb364d8" />
<img width="500" alt="Image" src="https://github.com/user-attachments/assets/4c61f72f-2a11-4c1c-88b3-368ab0f9b504" />

**Comment by fredrikekelund** [comment]
We could add some details about our offerings to the empty state page while we're at it. 

The fact that we offer Professional Email (Titan) and Google Workspace at X price is valuable information for users who are seeing the empty state, too.

**Comment by StevenDufresne** [comment]
Yeah, I'm up for a deeper cut on the empty state. Let's brainstorm!

For this page, I think it's about:
1. Explaining why (they should get personalized emails, and the necessary domain)
2. Explaining how

Domain purchases in this context are not usually impulsive so it's important to explain the value of domains and emails and surface the process as much as possible to remove any mental barriers for our users. If they are ready to purchase now great, we have a flow for them, if they buy a domain later, they'll know what to do next.

Here's a strategically rough and ugly version of what I'm thinking when the user has no domain or plan:

<img width="1527" alt="Image" src="https://github.com/user-attachments/assets/d4c5a4ff-5b09-4bee-95d2-c800636c5143" />
 <br>
<br>
 If it seems like a good direction, I can refine the UI with a designer and talk to customers who purchased professional email to make the copy more compelling and tuned into real use cases.
